File: backend/src/services/historyService.js
const sessionHistory = []
let nextId = 1

export function listHistory() {
  // 前端历史列表默认想看到最新记录，所以这里直接按“最新在前”返回。
  return [...sessionHistory].reverse()
}

export function addHistoryEntry(entry) {
  // 当前历史记录只存在内存里，没有落数据库。
  // 所以后端服务一重启，历史就会清空。这是当前版本的刻意简化。
  const record = {
    id: nextId++,
    mode: entry.mode,
    modeLabel: entry.modeLabel,
    question: entry.question,
    reply: entry.reply,
    createdAt: new Date().toISOString(),
  }

  sessionHistory.push(record)

  return record
}

export function clearHistory() {
  sessionHistory.length = 0
}

File: backend/app/services/history_service.py
from datetime import datetime, timezone

from sqlalchemy import or_

from ..extensions import db
from ..models import ConversationHistory


def list_history(keyword: str = "", mode: str = "", limit: int = 50, user_id: int | None = None) -> list[dict]:
    query = ConversationHistory.query.filter(ConversationHistory.user_id == user_id)

    if mode:
        query = query.filter(ConversationHistory.mode == mode)

    if keyword:
        like_keyword = f"%{keyword.strip()}%"
        query = query.filter(
            or_(
                ConversationHistory.question.ilike(like_keyword),
                ConversationHistory.reply.ilike(like_keyword),
            )
        )

    records = (
        query.order_by(ConversationHistory.created_at.desc())
        .limit(max(1, min(limit, 200)))
        .all()
    )
    return [record.to_dict() for record in records]


def add_history_entry(entry: dict) -> dict:
    record = ConversationHistory(
        user_id=entry.get("userId"),
        mode=entry["mode"],
        mode_label=entry["modeLabel"],
        question=entry["question"],
        reply=entry["reply"],
        created_at=datetime.now(timezone.utc),
    )
    db.session.add(record)
    db.session.commit()
    return record.to_dict()


def clear_history(user_id: int | None = None) -> None:
    ConversationHistory.query.filter(ConversationHistory.user_id == user_id).delete()
    db.session.commit()


def delete_history_entry(record_id: int, user_id: int | None = None) -> None:
    record = ConversationHistory.query.filter_by(id=record_id, user_id=user_id).first()

    if not record:
        raise ValueError("历史会话不存在")

    db.session.delete(record)
    db.session.commit()
